Provide basic connected containerized installation (#4750)
theforeman/foreman-documentation: guides/common/modules/proc_deploy-project-server-configuration-with-the-default-database.adoc

:_mod-docs-content-type: PROCEDURE

[id="deploy-{project-context}-server-configuration-with-the-default-database"]
= Deploy {ProjectServer} configuration with the default database

[role="_abstract"]
You can deploy a {ProjectServer} in a default configuration by using the `{foremanctl-installer} deploy` command.

> You can extend the deployment by selecting one or more features or adding one or more configuration options.

Reviewer: Nitpick trigger warning:
```suggestion
To extend the default deployment, you can select one or more features or add one or more configuration options.
```
I just disliked that both sentences started with "You can" :) It felt off. Not a blocker.

The initial deployment also installs PostgreSQL databases on the same server.

[NOTE]
====

> Any additional run of the `{foremanctl-installer} deploy` command overwrites the corresponding configuration options and records them in the {Project} answer file.

Reviewer 2: The answer file is related to the Puppet installer used in the "old" satellite-installer. We now have a `parameters.yaml` file.

Author: So just this? Or is there more that's changed in principle?

```suggestion
Any additional run of the `{foremanctl-installer} deploy` command overwrites the corresponding configuration options and records them in the `parameters.yaml` file.
```

Reviewer 2: Yes

> This way, you can restore a broken system or add additional settings to your deployment.

Reviewer 2: > This way, you can restore a broken system or add additional settings to your deployment.

Should we mention this here? IMO, a separate section on _How to restore a broken system_ is a better place.

Author: A dedicated restore/recovery section could work. But would it be more complex than just "run installer again"? If not, I would keep it as is.

Reviewer 2: The thing is, and it's probably out of the scope for this PR, that not all of the parameters are persisted, meaning the parameters.yaml contains _some_ of them, not all of them.
I'm okay with keeping it as it is and talking about the restore/recovery later.

====

.Prerequisites
* Your system conforms with the requirements.
For more information, see xref:{InstallingServerDocURL}planning-{project-context}-server-installation_{project-context}[Planning {ProjectServer} installation].
* Your system is prepared for the installation.
For more information, see xref:{InstallingServerDocURL}preparing-environment-for-{project-context}-server-installation_{project-context}[Preparing environment for {ProjectServer} installation].
* Your system has the `:project-installer-package:` package installed.

.Procedure
. Update the system to the latest available version:
+
[options="nowrap" subs="+quotes,attributes"]
----
# {package-upgrade}
----
. Optional: Inspect the features available for deployment:
+
[options="nowrap" subs="+quotes,attributes"]
----
# {foremanctl-installer} features
----
+
You can enable one or more features by adding the `--add-feature _Feature-1_ _Feature-2_` option to the `{foremanctl-installer} deploy` command.
. Deploy the configuration on the server:
+
[options="nowrap" subs="+quotes,attributes"]
----

> # {foremanctl-installer} deploy\

Reviewer 2: ```suggestion
# {foremanctl-installer} deploy \
```

--foreman-initial-admin-username _My_admin_user_name_ \
--foreman-initial-admin-password _My_admin_password_
----
+
The script displays its progress and writes logs to `{installer-log-file}`.

ifdef::satellite[]

> ifeval::["{mode}" == "disconnected"]

Reviewer 2: Note to myself: What's the expected behavior for disconnected installations?

Author: Should I just remove this section for now? This PR is intended for just connected installs, but I left this here because it would not show up in the connected build anyways.

Reviewer 2: Keep it here, and we can talk about it later in the refinement.

:_mod-docs-content-type: PROCEDURE

[id="install-a-basic-{project-context}-server"]
= Install a basic {ProjectServer}

[role="_abstract"]
You can install a {ProjectServer} in a default configuration by using the `{foremanctl-installer} deploy` command.
You can customize the installation by selecting one or more features or adding one or more configuration options.

This initial configuration creates an organization, location, user name, and password.
The initial configuration also installs PostgreSQL databases on the same server.

[NOTE]
====
Any additional run of the `{foremanctl-installer} deploy` command overwrites the corresponding configuration options and records them in the {Project} answer file.
This way, you can restore a broken system or add additional settings to your deployment.
//For more information on how to apply custom configuration on other services, see {InstallingServerDocURL}applying-custom-configuration_{project-context}[Applying Custom Configuration to {Project}].

> ====

Reviewer: We should try to make sure we don't add too many admonitions so I wonder if this would work as a paragraph after the procedure -- as part of the last step. As a sort of summary of the installation process results? Or perhaps even rephrase it as `.Next step`?

Author: Agree, but we are about to remove the second admonition, which should leave just this one.

I wanted to keep it early, because this is really more of a conceptual piece of information that I think works better beforehand, so that the user knows the deployment can be incremental.
